csrc/expr_simplifier.cpp

```diff
@@ -1557,11 +1541,9 @@ bool lessThan(Val* x, Val* y, const Context& context) {
 bool lessEqual(Val* x, Val* y, const Context& context) {
   x = foldConstants(x);
   y = foldConstants(y);
-  if (x->getInt().has_value() && y->getInt().has_value()) {
-    return *x->getInt() <= *y->getInt();
-  }
-  if (x->getDouble().has_value() && y->getDouble().has_value()) {
-    return *x->getDouble() <= *y->getDouble();
+  if (x->value().hasValue() && x->value().is<int64_t>() &&
+      y->value().hasValue() && y->value().is<int64_t>()) {
```

Comment:
I think this needs to be
```suggestion
  if (x->value().hasValue() && y->value().hasValue()) {
```